From f79c274f0201d16097535397378ae08faf37ae05 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Fri, 12 Jan 2024 17:51:21 +0100
Subject: [PATCH 0823/1141] media: dcmipp: bytecap: call mutex_destroy on
 release and error handling

Add a mutex_destroy during the error handling within the function
dcmipp_bytecap_ent_init and during the dcmipp_bytecap_ent_release

Change-Id: Ia5ef4537ff91b99e571eab5f024c84ffdc10c9ac
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/351137
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
---
 .../media/platform/st/stm32/stm32-dcmipp/dcmipp-bytecap.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-bytecap.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-bytecap.c
@@ -791,6 +791,7 @@ void dcmipp_bytecap_ent_release(struct d
 	struct dcmipp_bytecap_device *vcap =
 		container_of(ved, struct dcmipp_bytecap_device, ved);
 
+	mutex_destroy(&vcap->lock);
 	media_entity_cleanup(ved->ent);
 	vb2_video_unregister_device(&vcap->vdev);
 }
@@ -995,7 +996,7 @@ struct dcmipp_ent_device *dcmipp_bytecap
 	if (ret) {
 		dev_err(dev, "%s: vb2 queue init failed (err=%d)\n",
 			entity_name, ret);
-		goto err_clean_m_ent;
+		goto err_mutex_destroy;
 	}
 
 	/* Initialize buffer list and its lock */
@@ -1035,12 +1036,13 @@ struct dcmipp_ent_device *dcmipp_bytecap
 	if (ret) {
 		dev_err(dev, "%s: video register failed (err=%d)\n",
 			vcap->vdev.name, ret);
-		goto err_clean_m_ent;
+		goto err_mutex_destroy;
 	}
 
 	return &vcap->ved;
 
-err_clean_m_ent:
+err_mutex_destroy:
+	mutex_destroy(&vcap->lock);
 	media_entity_cleanup(&vcap->vdev.entity);
 err_clean_pads:
 	dcmipp_pads_cleanup(vcap->ved.pads);
